layer-atlas/src/main/java/com/layer/atlas/AtlasAvatar.java

```diff
@@ -175,66 +174,37 @@ public AtlasAvatar setParticipants(Set<Identity> participants) {
     private void update() {
         // Limit to MAX_AVATARS valid avatars, prioritizing participants with avatars.
         if (mParticipants.size() > MAX_AVATARS) {
-            Queue<Identity> withAvatars = new LinkedList<>();
-            Queue<Identity> withoutAvatars = new LinkedList<>();
-            for (Identity participant : mParticipants) {
-                if (participant == null) continue;
-                if (!TextUtils.isEmpty(participant.getAvatarImageUrl())) {
-                    withAvatars.add(participant);
-                } else {
-                    withoutAvatars.add(participant);
-                }
-            }
-
+            List<Identity> maxAvatars = Observable.fromIterable(mParticipants)
+                    .skip(mParticipants.size() - MAX_AVATARS)
+                    .toList().blockingGet();
             mParticipants = new LinkedHashSet<>();
-            int numWithout = Math.min(MAX_AVATARS - withAvatars.size(), withoutAvatars.size());
-            for (int i = 0; i < numWithout; i++) {
-                mParticipants.add(withoutAvatars.remove());
-            }
-            int numWith = Math.min(MAX_AVATARS, withAvatars.size());
-            for (int i = 0; i < numWith; i++) {
-                mParticipants.add(withAvatars.remove());
-            }
+            mParticipants.addAll(maxAvatars);
         }
 
         Diff diff = diff(mInitials.keySet(), mParticipants);
         List<ImageTarget> toLoad = new ArrayList<>();
 
-        List<ImageTarget> recyclableTargets = new ArrayList<ImageTarget>();
         for (Identity removed : diff.removed) {
             mInitials.remove(removed);
             ImageTarget target = mImageTargets.remove(removed);
             if (target != null) {
                 mPicasso.cancelRequest(target);
-                recyclableTargets.add(target);
             }
         }
 
+        mImageTargets.clear();
+        mInitials.clear();
         for (Identity added : diff.added) {
             if (added == null) return;
             mInitials.put(added, Util.getInitials(added));
 
             final ImageTarget target;
-            if (recyclableTargets.isEmpty()) {
-                target = new ImageTarget(this);
-            } else {
-                target = recyclableTargets.remove(0);
-            }
+            target = new ImageTarget(this);
```

Comment:
define and initialize variable in one line `final ImageTarget target = new ImageTarget(this)`

Reply:
👍